napari/_vispy/utils/gl.py

```diff
@@ -96,14 +97,17 @@ def fix_data_dtype(data):
         return data
 
     try:
-        dtype_ = {
-            "i": np.float32,
-            "f": np.float32,
-            "u": np.uint16,
-            "b": np.uint8,
-        }[dtype.kind]
+        dtype_ = cast(
+            'type[np.dtype[Any]]',
+            {
+                "i": np.float32,
+                "f": np.float32,
+                "u": np.uint16,
+                "b": np.uint8,
+            }[dtype.kind],
+        )
         if dtype_ == np.uint16 and dtype.itemsize > 2:
-            dtype_ = np.float32
+            dtype_ = np.float32  # type: ignore [assignment]
```

Comment:
why is this necessary btw?

Reply:
Imho a problem in mypy type inference:

> error: Incompatible types in assignment (expression has type "type[floating[Any]]", variable has type "type[dtype[Any]]")  [assignment]

I tried various combinaison above, to no success, and AFAIC, I dont se why `X[floating[Y]]` can't be assinged to `X[dtype[Y]]`

Reply:
wait, maybe I can try a union. Also maybe we can refactor the try-except with `dtype.kind in 'ifub'` ?

Reply:
Ok, find a way to do it with a union. That does not get rid of the cast, but it's better. thanks